### PR TITLE
Add dynamic shape test for unsqueeze_copy

### PR DESCRIPTION
**Pull request by vanbasten23**

### CODE REVIEW

File: torch_xla/csrc/tensor_methods.cpp [hunk 1]
```diff
@@ -2567,10 +2567,10 @@ void uniform_(XLATensorPtr& input, double from, double to) {
 }
 
 XLATensorPtr unsqueeze(const XLATensorPtr& input, int64_t dim) {
-  auto input_shape = input->shape();
+  xla::util::MaybeRef<xla::Shape> input_shape = input->shape();
```
Comment (alanwaketan) [inline torch_xla/csrc/tensor_methods.cpp:2570]: Why this would fix anything?

File: test/test_dynamic_shapes.py [hunk 2]
```diff
@@ -182,6 +182,30 @@ def test_expand_symint_correctness(self):
     self.assertEqual(t3.shape[0], 2)
     self.assertEqual(expand_out_aten.cpu(), expand_out_xla.cpu())
 
+  def test_unsqueeze_copy_dynamism(self):
+    t1 = torch.tensor([[1, 0, 0, 5, 0, 6], [1, 3, 2, 0, 0, 1]], device=dev)
+    t2 = torch.nonzero(t1)
+    # t2.shape=torch.Size([<=12, 2]) with real size [7, 2]
+
```
Comment (JackCaoG) [inline test/test_dynamic_shapes.py:189]: nit: remove empty line

**Review by JackCaoG — APPROVED** [review]
minor nits, otherwise lgtm

File: torch_xla/csrc/tensor_methods.cpp (hunk 1 above)
Comment (vanbasten23) [reply torch_xla/csrc/tensor_methods.cpp:2570]: It doesn't fix anything. I find it easier to read the code when I make the type explicit.

Comment (alanwaketan) [reply torch_xla/csrc/tensor_methods.cpp:2570]: hahaha, then you must hate python 😜 It's up to you. I'm fine with it.

**Review by alanwaketan — APPROVED** [review]
LGTM.

**Comment by vanbasten23** [comment]
Thanks for the review!

**Pull request merged**